ui/packages/consul-acls/app/components/consul/token/selector/index.hbs

```diff
@@ -0,0 +1,158 @@
+                    @disabled={{authForm.disabled}}
+                    @onchange={{authForm.submit}}
+                    @onerror={{authForm.error}}
+                  />
+                </authForm.Method>
+              {{/if}}
+            </AuthForm>
+          </BlockSlot>
+          <BlockSlot @name="actions">
+            <Action
+              {{on "click" modal.close}}
+            >
+              Continue without logging in
+            </Action>
+          </BlockSlot>
+        </ModalDialog>
+      </:unauthorized>
+      <:authorized as |authDialog|>
+        <ModalDialog
+          @name="login-toggle"
+          @onclose={{this.close}}
+          @onopen={{this.open}}
+          @aria={{hash
+            label="Log in with a different token"
+          }}
+        as |modal|>
+          <Ref
+            @target={{this}}
+            @name="modal"
+            @value={{modal}}
+          />
+          <BlockSlot @name="header">
+            <h2>
+              Log in with a different token
+            </h2>
+          </BlockSlot>
+          <BlockSlot @name="body">
+            <AuthForm
+              @dc={{@dc.Name}}
+              @nspace={{@nspace}}
+              @partition={{@partition}}
+              @onsubmit={{action authDialog.login value="data"}}
+            as |authForm|>
+              <Ref 
+                @target={{this}}
+                @name="authForm"
+                @value={{authForm}}
+              />
+            </AuthForm>
+          </BlockSlot>
+          <BlockSlot @name="actions">
+            <Action
+              {{on 'click' modal.close}}
+            >
+              Continue without logging in
+            </Action>
+          </BlockSlot>
+        </ModalDialog>
+        <Portal @target="app-before-skip-links">
+          <Action
+            {{on "click" (optional authDialog.logout)}}
+          >
+            Logout
+          </Action>
+        </Portal>
+        <PopoverMenu @position="right" as |components api|>
+          <BlockSlot @name="trigger">
+            Logout
+          </BlockSlot>
+          <BlockSlot @name="menu">
+            {{#let components.MenuItem components.MenuSeparator as |MenuItem MenuSeparator|}}
+{{!TODO: It might be nice to use one of our recursive components here}}
+{{#if @token.AccessorID}}
+                <li role="none">
+                  <AuthProfile
+                    @item={{@token}}
+                  />
+                </li>
+                <MenuSeparator />
+{{/if}}
+                <MenuItem
+                  class="dangerous"
+                  @onclick={{action authDialog.logout}}
+                >
+                  <BlockSlot @name="label">
+                    Logout
+                  </BlockSlot>
+                </MenuItem>
+              {{/let}}
+          </BlockSlot>
+        </PopoverMenu>
+      </:authorized>
+    </AuthDialog>
+
+  </li>
+{{yield
+  (hash
+    modal=this.modal
+  )
+}}
+{{/if}}
+
```

Comment:
Again naming this `Consul::Token::Selector` is consistent here, and at a stretch we are 'selecting' which token to use when we log in, so it just about works.